dex/wait/queue.go

```diff
@@ -193,7 +193,9 @@ func (q *TaperingTickerQueue) Run(ctx context.Context) {
 	defer wg.Wait()
 
 	runWaiter := func(w *taperingWaiter) {
+		wg.Add(1)
```

Comment:
You always `Add` before starting the goroutine, not at the beginning of the goroutine you are starting.

Reply:
Ah, right, thanks for calling this out, thought I'd simplify it but instead I got caught :)

Relevant doc section for `WaitGroup.Add`
```
// Note that calls with a positive delta that occur when the counter is zero
// must happen before a Wait. Calls with a negative delta, or calls with a
// positive delta that start when the counter is greater than zero, may happen
// at any time.
// Typically this means the calls to Add should execute before the statement
// creating the goroutine or other event to be waited for.
```

Fixed.